File: java/src/main/java/Gigasecond.java
import java.time.LocalDate;
import java.time.LocalDateTime;

public class Gigasecond {
    public static final int ONE_BILLION = 1_000_000_000;
    private final LocalDateTime gigasecond;

    public Gigasecond(final LocalDate moment) {
        this(moment.atTime(0,0));
    }

    public Gigasecond(final LocalDateTime moment) {
        gigasecond = moment.plusSeconds(ONE_BILLION);
    }

    public LocalDateTime getDateTime() {
        return gigasecond;
    }
}

File: scala/src/main/scala/Gigasecond.scala
import java.time.{LocalDate, LocalDateTime}
import java.time.temporal.ChronoUnit

object Gigasecond {
  def add(startDate: LocalDate): LocalDateTime = add(startDate.atStartOfDay)

  def add(startDateTime: LocalDateTime): LocalDateTime = startDateTime.plus(1_000_000_000L, ChronoUnit.SECONDS)
}
